src/components/v5/common/CompletedAction/partials/SetUserRoles/SetUserRoles.tsx

```diff
@@ -106,33 +108,37 @@ const SetUserRoles = ({ action }: Props) => {
     constraint: 'excludeInheritedRoles',
   });
 
-  const actionRoles = transformActionRolesToColonyRoles(roles);
-  // in case of motions, no historic roles are created so we just assume we are modifying their current roles (which contract wise we are)
+  const oldRoles = historicRoles?.getColonyHistoricRole
+    ? transformApiRolesToArray(historicRoles.getColonyHistoricRole)
+    : currentUserRoles;
 
-  const dbPermissionsOld =
-    actionRoles.filter(Boolean).length > 0 // if we didnt just remove all the roles
-      ? actionRoles
-      : currentUserRoles;
+  if (!roles) {
+    console.warn('No roles present in action');
+    return null;
+  }
 
-  const { role: dbRoleForDomainOld } = getRole(
-    dbPermissionsOld,
-    areRolesMultiSig,
-  );
+  const normalizedRoles = normalizeRolesForAction(roles);
+  const isRemovingRoles =
+    normalizedRoles.filter((role) => role.setTo).length === 0;
 
-  const dbPermissionsNew = transformActionRolesToColonyRoles(
-    historicRoles?.getColonyHistoricRole || roles,
-    {
-      isMotion: !!motionData || !!multiSigData,
-    },
-  );
+  /*
+   * This probably seems so confusing but dropping a here be dragons comment for future developers
+   * When you create a motion, there is no historic roles created, that's created when a setUserRoles contract passes
+   * when you do it via permissions, it 100% reflects the roles the user currently holds
+   * The roles on the action are just a diff, so if we add just 1 role, only that one is there, the same when we remove it
+   * Motions however modify the user's current roles in the colony, so we use those
+   */
```

Comment:
Needed a couple of reads to actually understand it, but it totally makes sense 👍